Below is a complete Python tutorial on working with numbers, covering basic arithmetic, number types, conversions, mathematical functions, and formatting.

### 📌 Python Numbers Tutorial
### This tutorial will cover:

1. Types of Numbers in Python
2. Basic Arithmetic Operations
3. Order of Operations (PEMDAS)
4. Number Functions (abs(), pow(), round(), etc.)
5. Math Module (math)
6. Number Conversions (int, float, str)

1️⃣ Types of Numbers in Python

In [4]:
# Integer (whole number)
x = 10
print(type(x)) # Output: <class 'int'>

# float (Decimal number)
y = 3.14
print(type(y)) # Output: <class 'float'>

# Complex number (a + bj format)
z = 2 + 3j
print(type(z)) # Output: <class 'complex'>


<class 'int'>
<class 'float'>
<class 'complex'>


2️⃣ Basic Arithmetic Operations

In [36]:
a = 10
b = 3

print(a + b) # addition (10 + 3 = 1)
print(a - b) # substraction ( 10 - 3 = 7)
print(a * b) # Multiplication ( 10 * 3 = 30)
print(a / b) # division ( 10 / 3 = 3.3333)
print(a // b)  # floor division ( 10 //3 = 3)
print(a % b) # Module (reminder) ( 10 % 3 = 1)
print(a ** b) # Exponention (10^3 = 1000)
print(abs(-3.8)) # print abslut it shows 3.8
print(pow(a,b))  #Exponention (10^3 = 1000)
print(max(a,b)) # max number 10
print(round(3.7)) # round up to 4

13
7
30
3.3333333333333335
3
1
1000
3.8
1000
10
4


3️⃣ Order of Operations (PEMDAS) \
Python follows the PEMDAS rule for calculations:

1. Parentheses
2. Exponents
3. Multiplication
4. Division
5. Addition
6.Subtraction



In [13]:
result  = (5 +3) * 2 ** 2 /4 - 1
print(result) # Output : 7.0

7.0


5️⃣ Math Module (math) \
Python has a built-in math module for advanced calculations.

In [37]:
import math # ( note from math import *  # Import all functions from math)
print(math.sqrt(16)) # square root = 4.0
print(math.ceil(3.4)) # round up 4
print(math.floor(3.9)) # round down to 3
print(math.log(10)) # natural log of 10
print(math.sin(math.radians(30))) # sin of 30 degree
print(math.pi)  # value of pi
print(math.e) # value of e


4.0
4
3
2.302585092994046
0.49999999999999994
3.141592653589793
2.718281828459045


6️⃣ Number Conversions

In [25]:
# Conver string to number 
num_str = "100"
num_int = int(num_str) # convert to integer
num_float = float(num_str) # convert to float

print(num_int + 5) # Output : 105
print(num_float + 3.5) # Output: 102.5

# convert number to string
age = 30
age_str = str(age)
print("I am " + age_str + " years old.") # Output: I am 30 years old

105
103.5
I am 30 years old.


# 📌 Python Numbers Tutorial

## 🔹 Summary Table: Common Number Operations

| **Operation**      | **Symbol/Function** | **Example**         | **Output**  |
|-------------------|------------------|-----------------|-----------|
| **Addition**      | `+`               | `10 + 5`        | `15`      |
| **Subtraction**   | `-`               | `10 - 5`        | `5`       |
| **Multiplication**| `*`               | `10 * 5`        | `50`      |
| **Division**      | `/`               | `10 / 3`        | `3.333...` |
| **Floor Division**| `//`              | `10 // 3`       | `3`       |
| **Modulus**       | `%`               | `10 % 3`        | `1`       |
| **Exponentiation**| `**`              | `2 ** 3`        | `8`       |
| **Absolute Value**| `abs()`           | `abs(-5)`       | `5`       |
| **Power**        | `pow(x, y)`        | `pow(2, 3)`     | `8`       |
| **Rounding**      | `round()`         | `round(3.14159, 2)` | `3.14` |
| **Square Root**   | `math.sqrt()`     | `math.sqrt(16)` | `4.0`     |
| **Logarithm**     | `math.log(x)`     | `math.log(10)`  | `2.302`   |

---


